internal/controller/promotion/promotion.go

```diff
@@ -11,27 +11,26 @@ import (
 	yaml "sigs.k8s.io/yaml/goyaml.v3"
 
 	kargoapi "github.com/akuity/kargo/api/v1alpha1"
-	"github.com/akuity/kargo/internal/credentials"
+	"github.com/akuity/kargo/internal/controller/health"
 	"github.com/akuity/kargo/internal/expressions"
 	exprfn "github.com/akuity/kargo/internal/expressions/function"
 	"github.com/akuity/kargo/internal/kargo"
 )
 
-// PromotionStepRunner is an interface for components that implement the logic for
-// execution of the individual PromotionSteps of a user-defined promotion
-// process.
-type PromotionStepRunner interface {
-	// Name returns the name of the PromotionStepRunner.
+// StepRunner is an interface for components that implement the logic for
+// execution of an individual Step in a user-defined promotion process.
+type StepRunner interface {
+	// Name returns the name of the StepRunner.
 	Name() string
-	// RunPromotionStep executes an individual step of a user-defined promotion
-	// process using the provided PromotionStepContext. Implementations may
-	// indirectly modify that context through the returned PromotionStepResult to
-	// allow subsequent promotion steps to access the results of its execution.
-	RunPromotionStep(context.Context, *PromotionStepContext) (PromotionStepResult, error)
+	// Run executes an individual Step from a user-defined promotion process using
```

Comment:
Related to our discussions offline:

> Run executes [...]

makes me think I may not have been too far off with `StepExecutor`.